### PR TITLE
New Extension: MarginNote

### PR DESCRIPTION
**Pull request by ourongxing**
## Description

> Initial Version

It's used to enhance MarginNote.

## Screencast
Raycast 2022-11-11 at 17 21 56
Raycast 2022-11-11 at 17 22 01
Raycast 2022-11-11 at 17 22 05

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by ourongxing** [comment]
@pernielsentikaer How long will it take?

File: extensions/marginnote/package.json [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+          "description": "Tags that you usually use",
+          "type": "textfield",
+          "required": false,
+          "placeholder": "Just like #tag1 #tag2 #tag3"
+        }
+      ]
+    },
+    {
+      "name": "restart",
+      "title": "Restart MarginNote",
```
Comment (pernielsentikaer) [inline extensions/marginnote/package.json:130]:
```suggestion
      "title": "Restart Application",
```

File: extensions/marginnote/package.json [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      "subtitle": "MarginNote",
+      "description": "Restart marginote and skip the unsigned mnaddon alert",
```
Comment (pernielsentikaer) [inline extensions/marginnote/package.json:132]:
```suggestion
      "description": "Restart MarginNote and skip the unsigned mnaddon alert",
```
mnaddon?

File: extensions/marginnote/src/searchNotebook.tsx [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        content={`[${notebook.title}](marginnote3app://notebook/${notebook.id})`}
+      />
+    </ActionPanel>
+  )
+}
+
+const NotFound = () => {
+  return (
+    <Detail
+      markdown={`# ⚠️ Not found data source from MarginNote\n ## Please install MarginNote v3.7.21 or lastest and "Raycast Enhance" addon.`}
```
Comment (pernielsentikaer) [inline extensions/marginnote/src/searchNotebook.tsx:182]:
```suggestion
      markdown={`# ⚠️ Not found data source from MarginNote\n ## Please install MarginNote v3.7.21 or latest and "Raycast Enhance" addon.`}
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension and I have some feedback ready for you:

* `Seach Notebook` doesn't seem to work, or I may have misconfigured it.

image

* The `Restart` command doesn't check whether MarignNote is installed
* Where do I find the preference values from `Take Note`
* You might need to update the README making some of this a bit more clear.

CleanShot 2022-11-14 at 15 13 47@2x

I'll add more when this is fixed, since it's not possible to test it yet 😊

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by ourongxing** [comment]
1. `Search Notebook` need install a MarginNote Plugin. Try to Click `Enter` to Download.
2. Actually all commands require MarginNote to be installed, if someone installs this extension, I don't think it's necessary to detect if Margin Note is installed.
3. `Take Note` preference is just a note url, For regular users of MarginNote, it's easy to get.

**Comment by ourongxing** [comment]
> 
> * `Seach Notebook` doesn't seem to work, or I may have misconfigured it.
> 
1. `Search Notebook` need install a MarginNote Plugin. Try to Click `Enter` to Download.

> 
> * The `Restart` command doesn't check whether MarignNote is installed

2. Actually all commands require MarginNote to be installed, if someone installs this extension, I don't think it's necessary to detect if MarginNote is installed.


> * Where do I find the preference values from `Take Note`
> * You might need to update the README making some of this a bit more clear.

3. `Take Note` preference is just a note url, For regular users of MarginNote, it's easy to get.

**Comment by pernielsentikaer** [comment]
> Search Notebook need install a MarginNote Plugin. Try to Click Enter to Download.
Then I get this error

image

> Actually all commands require MarginNote to be installed, if someone installs this extension, I don't think it's necessary to detect if MarginNote is installed.

If something throws an error then we need to handle it gracefully 😊

> Take Note preference is just a note url, For regular users of MarginNote, it's easy to get.

So how do I get it?

**Comment by ourongxing** [comment]
> image

To install uncertified plug-ins in MarginNote, you need to turn on the switch in the figure. If the raycast extension passes, it will also be certified soon.
image


> If something throws an error then we need to handle it gracefully 😊

OKay, just wait for moment.

> So how do I get it?

OKay. Follow the picture.
image

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
It still doesn't work after installing the extension (and I also tried to restart MarginNote), how can I help you debug, you can always catch me in [Slack](https://www.raycast.com/community)

CleanShot 2022-11-15 at 09 09 50@2x

**Comment by ourongxing** [comment]
> It still doesn't work after installing the extension (and I also tried to restart MarginNote), how can I help you debug, you can always catch me in [Slack](https://www.raycast.com/community)
> 
> CleanShot 2022-11-15 at 09 09 50@2x

Maybe you need open a notebook first.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

It seems like I got the most working, and also updated to the latest enhancement version.

I'm not sure the "skip" alert is working, since the extension might be unloaded the second you open MarginNote due to this setting.

CleanShot 2022-11-16 at 05 44 30@2x

I think we should remove that from the code.

**Comment by ourongxing** [comment]
image

I think it still works. It just clicks `enter` when MarginNote opens

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**